### PR TITLE
Drawer Component

### PR DESCRIPTION
**Pull request by hobbescodes**
##### Task link: https://trello.com/c/iBERH60r/108-drawer-component

Added custom `Drawer` component with `trigger`, `placement`, `title`, and `description` props. Also added primitive exports to allow for other users to build and customize `Drawer` components outside of AR branding/usage.

NOTE: Also created custom `Icon` component in this PR to allow for ease of use with `react-icons` in storybook and complete components.

## Test Steps

1) Verify that everything renders appropriately in storybook
2) Verify that all tests pass
3) Verify that `Drawer` usage in downstream app works as expected
4) Verify that `primitives` module gets bundled appropriately, and that users may construct custom `Drawer` components in a downstream app.

### CODE REVIEW

File: src/components/client/core/Drawer/Drawer.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  render: () => (
+    <Drawer
+      trigger={<Button>Open Drawer</Button>}
+      title="Drawer Title"
+      description="Drawer Description"
+      // placement="left"
```
Comment (coopbri) [inline src/components/client/core/Drawer/Drawer.stories.tsx:14]: This can be removed

File: src/components/client/core/Drawer/recipe.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { sva } from "generated/panda/css";
```
Comment (coopbri) [inline src/components/client/core/Drawer/recipe.ts:1]: I think we should adopt the pattern of `[ComponentName].recipe.ts` (e.g. `Drawer.recipe.ts`) for recipe file names.

It looks like Panda CSS does this as well (example: https://panda-css.com/docs/concepts/slot-recipes#defining-the-recipe), though my preference would be to capitalize the name to embody the "React component" intent.

File: src/components/client/core/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  DrawerTrigger,
+} from "components/primitives";
+
+import type { DrawerProps } from "components/primitives";
+import type { ReactNode } from "react";
+
+export interface Props extends DrawerProps {
+  trigger: ReactNode;
+  // TODO: remove hard-coded values
+  placement?: "left" | "right";
```
Comment (coopbri) [inline src/components/client/core/Drawer/Drawer.tsx:23]: I think it's okay to keep this narrow type here and remove this TODO. I suppose you could extract it to an explicit type or even an enum if you'd like, but I think it's fine!

**Review by coopbri — APPROVED** [review]
Everything working great for me both on Storybook and in the example Next.js app. Love the patterns and architecture you established for the library here, should be really scalable. Thanks a lot!!

File: src/components/client/core/Drawer/Drawer.tsx (hunk 3 above)
Comment (hobbescodes) [reply src/components/client/core/Drawer/Drawer.tsx:23]: Ok sounds good! Yeah I think I will remove for now, if we decide later that explicit types or enums are better, we can go that route too!

**Review by Twonarly1 — APPROVED** [review]
Test passed and storybook looks good. Had trouble testing with downstream examples/next app. [Error] could not resolve @animareflection/ui.

**Pull request merged**